crates/astria-bridge-withdrawer/src/main.rs

```diff
@@ -41,20 +62,62 @@ async fn main() -> ExitCode {
         .wrap_err("failed to setup telemetry")
     {
         Err(e) => {
-            eprintln!("initializing conductor failed:\n{e:?}");
+            eprintln!("initializing bridge withdrawer failed:\n{e:?}");
             return ExitCode::FAILURE;
         }
         Ok(metrics_and_guard) => metrics_and_guard,
     };
 
     info!(
         config = serde_json::to_string(&cfg).expect("serializing to a string cannot fail"),
-        "initializing conductor"
+        "initializing bridge withdrawer"
     );
 
+    let (frost_participant_clients, frost_public_key_package) = if cfg.no_frost_threshold_signing {
+        (None, None)
+    } else {
+        let public_key_package = match read_frost_key(&cfg.frost_public_key_package_path)
+            .wrap_err_with(|| {
+                format!(
+                    "failed reading frost public key package from file `{}`",
+                    cfg.frost_public_key_package_path
+                )
+            }) {
+            Err(error) => {
+                error!(%error, "failed to read frost public key package");
+                return ExitCode::FAILURE;
+            }
+            Ok(key) => key,
+        };
+
+        let frost_participant_endpoints = cfg
+            .frost_participant_endpoints
+            .split(',')
+            .map(str::to_string)
+            .collect();
+        let participant_clients = match initialize_frost_participant_clients(
+            frost_participant_endpoints,
+            &public_key_package,
```

Comment:
I see what you did here. :D I should have specified my original comment better:

The parsing should happen inside the `signer::frost` module, ideally with a `FromStr for ParticipantEndpoints` (and with a simple unit test).

The initialization of the clients should happen when the run-loop is entered. I have originally suggested the `Startup` type, but that's wrong. It should be at the beginning of `Submitter::run` here (link to current main: https://github.com/astriaorg/astria/blob/a12efeb0e4000d8ac2adc4e70ced4954cfbbb94c/crates/astria-bridge-withdrawer/src/bridge_withdrawer/submitter/mod.rs#L79-L91

The idea is that if a shutdown signal is received we can abort initialization without having to wait 30 seconds.